### ISSUE TITLE
Parser errors on common Zsh constructs (for...in...do...done, ||)

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#347]
## Parser crashes on common Zsh constructs

Running the built binary against a real .zshrc surfaces multiple parser panics / "no prefix parse function" errors for tokens that are fundamental to Zsh:

```
Parser Error in ~/.zshrc: no prefix parse function for IN found
Parser Error in ~/.zshrc: no prefix parse function for DO found
Parser Error in ~/.zshrc: no prefix parse function for DONE found
Parser Error in ~/.zshrc: no prefix parse function for || found
```

## Impact

- Any script using `for x in ...; do ...; done` triggers four errors before the linter gives up.
- `||` (logical OR between commands) is universal — parser refuses to progress.
- The binary still reports some violations after the errors, suggesting partial recovery, but downstream passes are running on truncated ASTs. False negatives are guaranteed.

## Reproduction

```bash
go build -o /tmp/zshellcheck ./cmd/zshellcheck
/tmp/zshellcheck ~/.zshrc
```

Against a typical `~/.zshrc` with PATH guards like:

```zsh
for dir in $HOME/.local/bin $HOME/bin; do
    [[ -d "$dir" ]] && path=("$dir" $path)
done
```

## Expected

- `for ... in ... do ... done` parses to a `ForStatement` AST node.
- `||` between commands parses to a `LogicalOr` AST node (or similar).
- No parser errors for these Zsh-core constructs.

## Scope of fix

Likely requires new prefix-parse handlers in `pkg/parser/` for the `IN`, `DO`, `DONE` tokens (if they are surfaced by the lexer as standalone tokens rather than being consumed by the `for` statement handler), and binary-operator handlers for `||`.

## Priority

**High.** The linter cannot be called "production-ready" for v0.3.35 public beta if real-world .zshrc files trip parser errors on the first try.

## References

- `pkg/parser/parser.go`
- `pkg/lexer/lexer.go`
- `pkg/token/token.go`
